### PR TITLE
chore: Improve duration

### PR DESCRIPTION
**Pull request by KazuCocoa**
For https://github.com/appium/WebDriverAgent/pull/500

### CODE REVIEW

File: WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m [hunk 1]
```diff
@@ -254,11 +254,13 @@ - (BOOL)fb_pressButton:(NSString *)buttonName
             buildError:error];
   }
 
-  (duration < 0)
-    // https://developer.apple.com/documentation/xctest/xcuiremote/1627476-pressbutton
-    ? [[XCUIRemote sharedRemote] pressButton:remoteButton]
+  if (duration) {
     // https://developer.apple.com/documentation/xctest/xcuiremote/1627475-pressbutton
-    : [[XCUIRemote sharedRemote] pressButton:remoteButton forDuration:duration];
+    [[XCUIRemote sharedRemote] pressButton:remoteButton forDuration:duration.doubleValue];
```
Comment (KazuCocoa) [inline WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m:259]: I found this method accepted negative value, so WDA did not raise an error here

File: WebDriverAgentTests/IntegrationTests/XCUIDeviceHelperTests.m [hunk 2]
```diff
@@ -119,14 +119,16 @@ - (void)disabled_testUrlSchemeActivation
 - (void)testPressingUnsupportedButton
 {
   NSError *error;
-  XCTAssertFalse([XCUIDevice.sharedDevice fb_pressButton:@"volumeUpp" forDuration:0 error:&error]);
+  NSNumber *duration;
```
Comment (mykola-mokhnach) [inline WebDriverAgentTests/IntegrationTests/XCUIDeviceHelperTests.m]: I assume duration value should be initialized here and below

File: WebDriverAgentLib/Commands/FBCustomCommands.m [hunk 3]
```diff
@@ -243,10 +243,8 @@ + (NSDictionary *)processArguments:(XCUIApplication *)app
 + (id<FBResponsePayload>)handlePressButtonCommand:(FBRouteRequest *)request
 {
   NSError *error;
-  NSNumber *requestedDuration = request.arguments[@"duration"];
-  NSTimeInterval duration = (requestedDuration ? requestedDuration.doubleValue : -1);
   if (![XCUIDevice.sharedDevice fb_pressButton:(id)request.arguments[@"name"]
-                                   forDuration:duration
+                                   forDuration:(NSNumber *)request.arguments[@"duration"]
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Commands/FBCustomCommands.m:247]: would this typecast work if the argument is nil?

File: WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.h [hunk 4]
```diff
@@ -81,7 +81,7 @@ NS_ASSUME_NONNULL_BEGIN
  @param duration Duration in seconds. This argument works only on tvOS. On iOS, this value will be ignored.
  @return YES if the button has been pressed
  */
-- (BOOL)fb_pressButton:(NSString *)buttonName forDuration:(NSTimeInterval)duration error:(NSError **)error;
+- (BOOL)fb_pressButton:(NSString *)buttonName forDuration:(nullable NSNumber *)duration error:(NSError **)error;
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.h:89]: in the docstring it makes sense to mention that nil values applies single press, which equals to 0.005 seconds in XCTest

File: WebDriverAgentLib/Commands/FBCustomCommands.m (hunk 3 above)
Comment (KazuCocoa) [reply WebDriverAgentLib/Commands/FBCustomCommands.m:247]: yes, then here was `nil`

File: WebDriverAgentTests/IntegrationTests/XCUIDeviceHelperTests.m [hunk 5]
```diff
@@ -119,14 +119,28 @@ - (void)disabled_testUrlSchemeActivation
...
+                                            forDuration:nil
+                                                  error:&error]);
+  XCTAssertNil(error);
+}
+
+- (void)testPressingSupportedButtonNumber
+{
+  NSError *error;
+  XCTAssertTrue([XCUIDevice.sharedDevice fb_pressButton:@"home"
+                                            forDuration:[NSNumber numberWithDouble:1.0]
```
Comment (mykola-mokhnach) [inline WebDriverAgentTests/IntegrationTests/XCUIDeviceHelperTests.m:142]: Could also be @(1.0)

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**